apps/backend/src/libs/modules/controller/base-controller.module.ts

```diff
@@ -24,13 +24,14 @@ class BaseController implements Controller {
 	public addRoute<HandlerOptions extends APIHandlerOptions = APIHandlerOptions>(
 		options: ControllerRouteParameters<HandlerOptions>,
 	): void {
-		const { handler, path } = options;
+		const { handler, path, preHandlers = [] } = options;
 		const fullPath = this.apiUrl + path;
 
 		this.routes.push({
 			...options,
 			handler: (request, reply) => this.mapHandler(handler, request, reply),
 			path: fullPath,
+			preHandlers,
```

Comment:
It should have the same logic as the handler, we want to control the information we pass to it, not the whole req/reply


```suggestion
			preHandlers:  preHandlers.map((preHandler) => {
				return (request, _, done): void => {
					this.mapPreHandler(preHandler, request, done);
				};
			}),
```